src/Transaction/AutoTransact.php

```diff
@@ -33,8 +33,7 @@ public function write(Mapper $mapper, string $method, Record $record) : void
             $this->commit();
         } catch (Exception $e) {
             $this->rollBack();
-            $c = get_class($e);
-            throw new $c($e->getMessage(), $e->getCode(), $e);
```

Comment:
I think probably casting to int will help here.

```
public __construct ([ string $message = "" [, int $code = 0 [, Throwable $previous = NULL ]]] )
```

taken from https://www.php.net/manual/en/exception.construct.php .

So the order of exception seems not changing. Also if this code is removed, that may affect other parts for this can become a BC break for people who rely on multiple exception. ( Just a thought )